sentry-core/src/main/java/io/sentry/core/Hub.java

```diff
@@ -18,52 +39,122 @@ public boolean isEnabled() {
 
   @Override
   public SentryId captureEvent(SentryEvent event) {
-    return null;
+    SentryId sentryId;
+    synchronized (lock) {
+      StackItem item = stack.peek();
+      sentryId = item.client.captureEvent(event, item.scope);
+      item.scope.setLastEventId(sentryId);
```

Comment:
If access through `Sentry.pushScope` for example, a new Hub instance will exist per thread. So it wouldn't be possible for a new scope to be pushed while another thread is executing line 45, before setting the last event Id here.
It could be a problem is a user creates a instance of a Hub and passes it around. In that case `lastEventId` could potentially be set on the wrong scope.

@HazAT, @mitsuhiko  is this just something we accept the risk and mitigate with docs? Or do we need to account for and lock `client.captureEvent + scope.setLastEventId`?

Reply:
For other SDKs `lastEventId` lives directly on the Hub,
So even there we can have race conditions which is fine.

Reply:
Things that usually don't go together are race condition and 'this is fine'.
But I won't fight anyone.
For as long as it doesn't blow up I don't care right here. 
Concurrent collection is ok. RWlock, full lock would give contention under high load though

Reply:
Given that a single `Hub` is _not supposed_ to be used by more than 1 thread I would not optimize for correctness of calls to `Hub.something` to be atomic with regards to client and scope calls but rather just make sure it is thread-safe. Meaning calls to Hub done concurrently will not throw.

That said, IMO a immutable or thread-safe collection is good enough here.

Reply:
Agreed offline to move lastEventId to the Hub instead of the scope.